cpp/src/Ice/LoggerAdminI.cpp

```diff
@@ -700,10 +686,9 @@ LoggerAdminLoggerI::log(const LogMessage& logMessage)
     {
         lock_guard lock(_mutex);
 
-        if(!_sendLogThread)
+        if(_sendLogThread.get_id() == thread::id())
```

Comment:
You use ".joinable()" pretty much everywhere else to check if a thread is initialized. Why get_id() here?

Reply:
This is to check if the thread has been initialized:

```
std::thread().get_id() == thread::id()
```

That is the case for a default constructed std::thread, which doesn't represent a thread. `joinable()` can be true if already finished.

Reply:
> I was wrong in the comment above, is still joinable until is joined.